lib/datadog/tracing/contrib/aws/instrumentation.rb

```diff
@@ -60,6 +64,98 @@ def annotate!(span, context)
+            when 'dynamodb'
+              add_dynamodb_tags(span, params)
+            when 'kinesis'
+              add_kinesis_tags(span, params)
+            when 'eventbridge'
+              add_eventbridge_tags(span, params)
+            when 'states'
+              add_states_tags(span, params)
+            when 's3'
+              add_s3_tags(span, params)
+            end
+          end
+
+          def add_sqs_tags(span, params)
+            queue_url = params[:queue_url]
+            queue_name = params[:queue_name]
+            # Regular expression to match the SQS queue URL pattern
+            # https://sqs.sa-east-1.amazonaws.com/12345678/MyQueueName
+            pattern = %r{https://sqs\.[^/]+\.amazonaws\.com/(\d+)/([^/]+)}
+
+            if queue_url && (match = queue_url.match(pattern))
+              aws_account, queue_name = match.captures
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_QUEUE_NAME, queue_name)
+          end
+
+          def add_sns_tags(span, params)
+            topic_arn = params[:topic_arn]
+            topic_name = params[:name]
+            if topic_arn
+              # example topic_arn: arn:aws:sns:us-west-2:123456789012:my-topic-name
+              topic_name = topic_arn.split(':')[-1]
+              aws_account = topic_arn.split(':')[-2]
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_TOPIC_NAME, topic_name)
+          end
+
+          def add_dynamodb_tags(span, params)
+            table_name = params[:table_name]
+            span.set_tag(Ext::TAG_TABLE_NAME, table_name)
+          end
+
+          def add_kinesis_tags(span, params)
+            stream_arn = params[:stream_arn]
+            stream_name = params[:stream_name]
+            if stream_arn
+              # example stream_arn: arn:aws:kinesis:us-east-1:123456789012:stream/my-stream
+              stream_name = stream_arn.split('/')[-1]
+              aws_account = stream_arn.split(':')[-2]
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_STREAM_NAME, stream_name)
+          end
+
+          def add_eventbridge_tags(span, params)
+            rule_name = params[:name] || params[:rule]
+            span.set_tag(Ext::TAG_RULE_NAME, rule_name)
+          end
+
+          def add_states_tags(span, params)
+            state_machine_name = params[:name]
+            state_machine_arn = params[:state_machine_arn]
+            execution_arn = params[:execution_arn]
+
+            if execution_arn
+              # 'arn:aws:states:us-east-1:123456789012:execution:example-state-machine:example-execution'
+              state_machine_name = execution_arn.split(':')[-2]
+            end
+
+            if state_machine_arn
+              # example statemachinearn: arn:aws:states:us-east-1:123456789012:stateMachine:MyStateMachine
```

Comment:
Should we be leveraging `ArnParser` from the gem?

https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/lib/aws-sdk-core/arn_parser.rb

Reply:
I tried using arn parser but it just made things more complex as it assumes all arns have 6 parts or at least splits that way thus causing `resourcename:stateMachine:MyStateMachine` meaning we would need to split again (not to mention the splitting they already do in the arn parser parse function. Additionally arn parser was not available on the aws sdk v2 (at least the one we were using for tests) so I would of had to check if the gem was defined dynamically and use a custom one if not. 
TLDR: Its seems to be for the better to use our own string parser regardless